src/ergw_aaa_diameter_srv.erl

```diff
@@ -429,11 +429,37 @@ filter_avp(AVPs, _) ->
     AVPs.
 
 check_avp_conditions(AVPs, [{AVP, Value} | Rest]) when is_map_key(AVP, AVPs) ->
-    case maps:get(AVP, AVPs) of
-	Value -> check_avp_conditions(AVPs, Rest);
+    case match_avp_value(Value, maps:get(AVP, AVPs)) of
+	true -> check_avp_conditions(AVPs, Rest);
 	_ -> false
     end;
 check_avp_conditions(_, []) ->
     true;
 check_avp_conditions(_, _) ->
     false.
+
+%% Since the implementation of not erlang native (json, yaml) configuration 
+%% format, we can not expect that the condition values will be matching the
+%% possible OTP diameter value formats. Some of the "variable" formats
+%% not likely to be used in matching (e.g. Time) are not implemented
+match_avp_value(ConditionVal, [AVPVal]) ->
+% OTP diameter puts message optional AVPs in a single element list
+    match_avp_value(ConditionVal, AVPVal);
+match_avp_value(ConditionVal, ConditionVal) ->
+    true;
+match_avp_value(ConditionVal, AVPVal)
+  when is_binary(ConditionVal) andalso is_integer(AVPVal) ->
+    AVPVal == binary_to_integer(ConditionVal);
+match_avp_value(ConditionVal, AVPVal)
+  when is_binary(ConditionVal) andalso is_atom(AVPVal) ->
+    AVPVal == binary_to_existing_atom(ConditionVal, utf8);
+match_avp_value(ConditionVal, AVPVal)
+  when is_binary(ConditionVal) andalso is_list(AVPVal) ->
+    AVPVal == binary_to_list(ConditionVal);
+match_avp_value(ConditionVal, AVPVal)
+  when is_binary(ConditionVal) andalso 
+       is_tuple(AVPVal) andalso (tuple_size(AVPVal) == 4 orelse tuple_size(AVPVal) == 8)->
+    {ok, AVPVal} == inet_parse:address(binary_to_list(ConditionVal));
```

Comment:
Might fail with badmatch. You have to have it inside ''when''.

Reply:
it is `==` not '=`